cumulusci/tasks/bulkdata/generate_mapping.py

```diff
@@ -160,60 +167,82 @@ def _build_mapping(self):
         """Output self.schema in mapping file format by constructing a dict and serializing to YAML"""
         objs = set(self.schema.keys())
         stack = self._split_dependencies(objs, self.refs)
+        ns = self.project_config.project__package__namespace
 
-        field_sort = (
-            lambda f: "  " + f
-            if f == "Id"
-            else (" " + f if f in self.core_fields else f)
-        )
+        def strip_namespace(element):
+            if self.options["strip_namespace"] and ns and element.startswith(f"{ns}__"):
+                return element[len(ns) + 2 :]
+            else:
+                return element
 
         self.mapping = {}
-        for obj in stack:
-            key = f"Insert {obj}"
+        for orig_obj in stack:
+            # Check if it's safe for us to strip the namespace from this object
+            stripped_obj = (
+                strip_namespace(orig_obj)
+                if strip_namespace(orig_obj) not in stack
+                else orig_obj
+            )
+            key = f"Insert {stripped_obj}"
             self.mapping[key] = {}
-            self.mapping[key]["sf_object"] = f"{obj}"
-            self.mapping[key]["table"] = f"{obj}"
+            self.mapping[key]["sf_object"] = stripped_obj
```

Comment:
My suggestion:

    stripped_obj = strip_namespace(orig_obj)
    obj = stripped_obj if stripped_obj not in stack else orig_obj
    key = f"Insert {obj}"
    self.mapping[key] = {}
    self.mapping[key]["sf_object"] = obj

Reply:
Adopted!